App.js

```diff
@@ -2,78 +2,34 @@ import { NavigationContainer } from '@react-navigation/native';
+import { Provider as PaperProvider } from 'react-native-paper';
 import { t } from 'react-native-tailwindcss';
-import {
-  createDrawerNavigator,
-  DrawerContentScrollView,
-  DrawerItemList,
-} from '@react-navigation/drawer';
 import useCachedResources from './hooks/useCachedResources';
-import BottomTabNavigator from './navigation/BottomTabNavigator';
-import LinkingConfiguration from './navigation/LinkingConfiguration';
+import OnboardingScreen from './screens/OnboardingScreen';
+import DrawerNavigator from './navigation/DrawerNavigator';
 
 const Stack = createStackNavigator();
-const Drawer = createDrawerNavigator();
-
-function Root() {
-  return (
-    <Stack.Navigator>
-      <Stack.Screen name="Root" component={BottomTabNavigator} />
-    </Stack.Navigator>
-  );
-}
-
-// Add drawer navigator header
-function CustomDrawerContent(props) {
-  return (
-    <DrawerContentScrollView {...props}>
-      <View style={[t.flex1, t.flexRow, t.p8, t.itemsCenter, t.bgGreen600]}>
-        <Headline style={[t.pR2]}>Username</Headline>
-        <Avatar.Icon size={24} icon="crown" />
-      </View>
-      <DrawerItemList {...props} />
-    </DrawerContentScrollView>
-  );
-}
 
 export default function App(props) {
   const isLoadingComplete = useCachedResources();
 
+  // Authentication flow set up pending
+  const isAuth = false;
+  // true: home screen ; false: login/signup onboarding screen
+  // need to manually RELOAD the app to update initialRoute
+
   if (!isLoadingComplete) {
     return null;
   }
   return (
     <PaperProvider>
       <View style={[t.flex1, t.bgWhite]}>
         {Platform.OS === 'ios' && <StatusBar barStyle="dark-content" />}
-        <NavigationContainer linking={LinkingConfiguration}>
-          <Drawer.Navigator
-            initialRouteName="Home"
-            drawerContent={(items) => <CustomDrawerContent {...items} />}
-          >
-            <Drawer.Screen
-              name="My Posts"
-              component={Root}
-              options={{
-                drawerIcon: () => <Avatar.Icon size={24} icon="folder-multiple-image" />,
-              }}
-            />
-            <Drawer.Screen
-              name="My Achievements"
-              component={Root}
-              options={{
-                drawerIcon: () => <Avatar.Icon size={24} icon="trophy" />,
-              }}
-            />
-            <Drawer.Screen
-              name="Logout"
-              component={Root}
-              options={{
-                drawerIcon: () => <Avatar.Icon size={24} icon="logout" />,
-              }}
-            />
-          </Drawer.Navigator>
+        <NavigationContainer>
+          <Stack.Navigator initialRouteName={isAuth ? 'Home' : 'Onboarding'} headerMode="none">
+            <Stack.Screen name="Onboarding" component={OnboardingScreen} />
+            <Stack.Screen name="Home" component={DrawerNavigator} />
+          </Stack.Navigator>
```

Comment:
with headerMode='none', phones with notches would make the screen look weird. Below might be a way to solve it
https://reactnavigation.org/docs/3.x/handling-iphonex/